### PR TITLE
Make inclusion of /v2/cql optional for REST API, with sysprop `stargate.rest.cql.disabled` (use Quarkus 2.16 `@EndpointDisabled`)

### LINKED ISSUE #2316: Add a way to opt-in to new `/v2/cql` endpoint (via REST API) -- using Quarkus `@EndpointDisabled` via `stargate.rest.cql.disabled`
(follow-up to #2313 )

Since it is possible that only some users may want to expose the new "raw CQL" endpoint for REST API, let's make its inclusion dynamic, based on a feature flag (Quarkus property that can be set via system props and/or env variables).

### PR DESCRIPTION
**Pull request by tatu-at-datastax**
**What this PR does**:

Makes the new CQL endpoint be optional included, depending on value of Boolean system property `stargate.rest.cql.enabled` (and its ENV variable counterpart, `STARGATE_REST_CQL_ENABLED`): value of `true` enables; any other value disables.
Default is that the endpoint is disabled (not included)

NOTE: should use new `@EndpointDisabled` Quarkus 2.16 adds, once 2.16.0 released (see https://github.com/quarkusio/quarkus/pull/30051 for details)

**Which issue(s) this PR fixes**:
Fixes #2316

**Checklist**
- [x] Changes manually tested
- [ ] Automated Tests added/updated
- [ ] Documentation added/updated
- [x] CLA Signed: [DataStax CLA](https://cla.datastax.com/)

### CODE REVIEW

**Comment by tatu-at-datastax** [comment]
Unfortunately approach does not work: CQL-endpoint tests should be failing now, they are not.
Looks like things do not work due to either bug or limitation wrt handling of `@LookupIfProperty` annotation.

I filed https://github.com/quarkusio/quarkus/issues/30022 to see if this might be considered a bug or just limitation.
Otherwise not sure if and how this could be achieved.

**Comment by tatu-at-datastax** [comment]
May decide to change to just fail at endpoint level -- but for now will not proceed until we know this is actually needed.

**Comment by ivansenic** [comment]
Let s wait for https://github.com/quarkusio/quarkus/pull/30051

**Comment by tatu-at-datastax** [comment]
Note: has to wait until https://github.com/quarkusio/quarkus/pull/30051 (now merged) gets released as part of Quarkus 2.16.0

File: apis/sgv2-restapi/CONFIGURATION.md [hunk 1]
```diff
@@ -0,0 +1,22 @@
+# Configuration Guide
+
+> **IMPORTANT:** Please check the [Stargate Common Configuration](../sgv2-quarkus-common/CONFIGURATION.md) for properties shared between all Stargate V2 APIs.
+
+## Stargate REST API Configuration
```
Comment (ivansenic) [inline apis/sgv2-restapi/CONFIGURATION.md:5]: is there no other restapi prop in the whole project?

**Review by ivansenic — APPROVED** [review]
Perfect :ship:

File: apis/sgv2-restapi/CONFIGURATION.md (hunk 1 above)
Comment (tatu-at-datastax) [reply apis/sgv2-restapi/CONFIGURATION.md:5]: Let me go check, good point.

Comment (tatu-at-datastax) [reply apis/sgv2-restapi/CONFIGURATION.md:5]: I don't see any, looking at `application.yaml`s under `sgv2-restapi`. There is at least one for SGv1 REST API (in-coordinator one) for inclusion of RESTv1 but nothing for SGv2 REST portion that I remember.

I'll keep this in mind if I mind something and make sure to update this page.

Comment (jeffreyscarpenter) [reply apis/sgv2-restapi/CONFIGURATION.md:5]: Make sure to also update the README, which currently says:

```
The REST API currently does not define any additional properties beyond these common properties.
```

Comment (tatu-at-datastax) [reply apis/sgv2-restapi/CONFIGURATION.md:5]: > Make sure to also update the README, which currently says:
> 
> ```
> The REST API currently does not define any additional properties beyond these common properties.
> ```

Changed. Thanks!

**Pull request merged**